backend/src/controllers/dataWarehouseController.js

```diff
@@ -96,6 +100,123 @@ export const createDataWareHouseJob = function createDataWareHouseJob() {
   );
 };
 
+export const findDataCube = async (req, res, next) => {
+  try {
+    const { query } = req;
+
+    if (query.v && query.operator && !Operator[query.operator]) {
+      return next(new InvalidRequest('The operations must be one of: ' + Object.keys(Operator).join(', ')));
+    }
+
+    const filter = {
+      ...(query.p ? { period: query.p } : {}),
+      ...(query.e ? { explorer: query.e } : {}),
+      ...(query.v && query.operator
+        ? {
+            totalSpent: {
+              [Operator[query.operator]]: query.v
+            }
+          }
+        : {})
+    };
+    const results = await dataCubeModel.find(filter);
+
+    res.json(results);
+  } catch (e) {
+    res.status(StatusCodes.INTERNAL_SERVER_ERROR).send(e.message);
+  }
+};
+
+export const createDataCube = async (req, res) => {
+  const lastYearAmount = 3;
+  const lastMonthsAmount = 36;
+  const years = Array.from(Array(lastYearAmount + 1).keys())
+    .slice(1)
+    .map(year => ({
+      keyword: `Y${String(year).padStart(2, '0')}`,
+      date: moment()
+        .subtract(year, 'years')
+        .toDate()
+    }));
+  const months = Array.from(Array(lastMonthsAmount + 1).keys())
+    .slice(1)
+    .map(month => ({
+      keyword: `M${String(month).padStart(2, '0')}`,
+      date: moment()
+        .subtract(month, 'months')
+        .toDate()
+    }));
+  const dates = years.concat(months);
+  const filter = dates
```

Comment:
Looks great! Maybe we should move the filter too?

Reply:
rt